string filename
string data
---
bool success
string message